test/test_datasets.py

```diff
@@ -480,6 +460,48 @@ def inject_fake_data(self, tmpdir, config):
         return num_images_per_category * len(categories)
 
 
+@unittest.skipIf(sys.platform in ('win32', 'cygwin'), 'temporarily disabled on Windows')
+class WIDERFaceTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.WIDERFace
+    FEATURE_TYPES = (PIL.Image.Image, (dict, type(None)))  # test split returns None as target
+    CONFIGS = datasets_utils.combinations_grid(split=('train', 'val', 'test'))
+
+    def inject_fake_data(self, tmpdir, config):
+        widerface_dir = pathlib.Path(tmpdir) / 'widerface'
+        annotations_dir = widerface_dir / 'wider_face_split'
+        os.makedirs(annotations_dir)
+
+        for split in ('train', 'val', 'test'):
+            split_idx = {
+                'train': 1,
+                'val': 2,
+                'test': 3,
+            }[split]
+
+            num_examples = 1
+            datasets_utils.create_image_folder(
+                root=tmpdir,
+                name=widerface_dir / f'WIDER_{split}' / 'images' / '0--Parade',
+                file_name_fn=lambda image_idx: f"0_Parade_marchingband_{image_idx + 1}_{split_idx}.jpg",
+                num_examples=num_examples,
+            )
+
+            annotation_file_name = {
+                'train': annotations_dir / 'wider_face_train_bbx_gt.txt',
+                'val': annotations_dir / 'wider_face_val_bbx_gt.txt',
+                'test': annotations_dir / 'wider_face_test_filelist.txt',
+            }[split]
```

Comment:
Same as above.

Reply:
I moved the `split_to_idx` out of the loop as it's needed in different places, but for this one it seems more natural to me to leave it here. Moving it out would mean declaring a new variable that's only used once (here), and I tend to prefer to declare things where they're used

Reply:
No strong opinion here. I'm aware that this completely up the alley of premature optimization, but for me it _feels_ wrong to declare this dictionary in every step and then simply select a single value.